File: archive/j/java/LeapYear.java
import java.util.*;

public class LeapYear {
    public static void main(String[] args) {
        int year;
        {
            Scanner sc = new Scanner(System.in);
            System.out.println("Enter year to check: ");
            year = sc.nextInt();

            if ((year % 4 == 0) && year % 100 != 0) {
                System.out.println(year + " is a leap year.");
            } else if ((year % 4 == 0) && (year % 100 == 0)
                    && (year % 400 == 0)) {
                System.out.println(year + " is a leap year.");
            } else {
                System.out.println(year + " is not a leap year.");
            }
        }
    }
}

File: archive/p/python/leapyear.py
def isLeapYear(n):
  if (n%100!=0 and n%4==0) or (n%100==0 and n%400==0):
    return True
  else:
    return False

year=int(input())
if isLeapYear(year)==True:
  print("Leap Year")
else:
  print("Not a leap year")
